content/library/api/connections/connections-sql.md

```diff
@@ -0,0 +1,18 @@
+---
+title: st.connections.SQL
+slug: /library/api-reference/connections/st.connections.sql
+---
+
+<Important>
+
+This is an experimental feature. Experimental features and their APIs may change or be removed at any time. To learn more, click [here](/library/advanced-features/prerelease#experimental-features).
+
+</Important>
+
+<Autofunction function="streamlit.connections.SQL" />
+
+<Autofunction function="streamlit.connections.SQL.query" />
+
+<Autofunction function="streamlit.connections.SQL.reset" />
+
+<Autofunction function="streamlit.connections.SQL.session" />
```

Comment:
Here's a markdown blurb I wrote about initializing SQL connections. @vdonato suggested it might be too long to put in the docstring.

---
Basic usage:
```python
import streamlit as st

conn = st.experimental_connection("sql")
df = conn.query("select * from pet_owners")
st.dataframe(df)
```

In case you want to pass a connection url (or other parameters) directly, it also works:

```python
conn = st.experimental_connection("local_db", type="sql", url="mysql://user:pass@localhost:3306/mydb")
```

Or specify parameters in secrets:

```toml
# .streamlit/secrets.toml
[connections.mydb]
dialect = "mysql"
username = "myuser"
password = "password"
host = "localhost"
database = "mydb"
```
```python
# streamlit_app.py
conn = st.experimental_connection("mydb", type="sql", autocommit=True)
```

As described above, some cloud databases use extra **kwargs to specify credentials.
These can be passed via secrets using `create_engine_kwargs` section.

```toml
# .streamlit/secrets.toml
[connections.snowflake]
url = "snowflake://<username>@<account>/"

[connections.snowflake.create_engine_kwargs.connect_args]
authenticator = "externalbrowser"
role = "..."
# ...
```

Reply:
Should this content be within the `content/library/api/connections/experimental-connection.md` file?

Reply:
Why does an example for `streamlit.connections.SQL` have content about `connections.snowflake`? Are you sure all of the above goes into the file: `content/library/api/connections/connections-sql.md` ??

Reply:
@sfc-gh-jcarroll where should the above content live?

Reply:
> Why does an example for streamlit.connections.SQL have content about connections.snowflake?

It's an example of a SQL connection to a Snowflake SQL database

> where should the above content live?

It could be below the examples and method information on this API reference page (similar to [this section](https://docs.streamlit.io/library/api-reference/performance/st.cache_data#using-streamlit-commands-in-cached-functions) in cache_data API page), OR we could weave it into the advanced features

I slightly prefer the API reference page but in theory I'm OK with the latter. If we decide to put it in advanced features, would love to link to that from API reference if possible.

Reply:
Done, I've added a Basic usage h3 section on this page via 125c823